src/molecule_plugins/azure/cookiecutter/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/create.yml

```diff
@@ -15,90 +15,98 @@
     keypair_path: "{{ lookup('env', 'MOLECULE_EPHEMERAL_DIRECTORY') }}/ssh_key"
   tasks:
     - name: Create resource group
-      azure_rm_resourcegroup:
+      azure.azcollection.azure_rm_resourcegroup:
         name: "{{ resource_group_name }}"
         location: "{{ location }}"
 
     - name: Create virtual network
-      azure_rm_virtualnetwork:
+      azure.azcollection.azure_rm_virtualnetwork:
         resource_group: "{{ resource_group_name }}"
         name: "{{ virtual_network_name }}"
         address_prefixes: "10.10.0.0/16"
 
     - name: Create subnet
-      azure_rm_subnet:
+      azure.azcollection.azure_rm_subnet:
         resource_group: "{{ resource_group_name }}"
         name: "{{ subnet_name }}"
         address_prefix_cidr: 10.10.1.0/24
         virtual_network_name: "{{ virtual_network_name }}"
 
     - name: Create key pair
-      user:
+      ansible.builtin.user:
         name: "{{ lookup('env', 'USER') }}"
         generate_ssh_key: true
         ssh_key_file: "{{ keypair_path }}"
       register: key_pair
 
     - name: Create molecule instance(s)
-      azure_rm_virtualmachine:
+      azure.azcollection.azure_rm_virtualmachine:
         resource_group: "{{ resource_group_name }}"
         name: "{{ item.name }}"
-        vm_size: Standard_A0
+        vm_size: Standard_D2_v4
         admin_username: "{{ ssh_user }}"
         public_ip_allocation_method: Dynamic
         ssh_password_enabled: false
+        managed_disk_type: Standard_LRS
         ssh_public_keys:
           - path: "/home/{{ ssh_user }}/.ssh/authorized_keys"
             key_data: "{{ key_pair.ssh_public_key }}"
         image:
-          offer: CentOS
-          publisher: OpenLogic
-          sku: '7.4'
+          offer: 0001-com-ubuntu-server-focal
+          publisher: Canonical
+          sku: 20_04-lts-gen2
           version: latest
       register: server
       with_items: "{{ molecule_yml.platforms }}"
       async: 7200
       poll: 0
 
     - name: Wait for instance(s) creation to complete
-      async_status:
+      ansible.builtin.async_status:
         jid: "{{ item.ansible_job_id }}"
       register: azure_jobs
       until: azure_jobs.finished
       retries: 300
       with_items: "{{ server.results }}"
+      notify:
+        - Populate instance config dict
+        - Convert instance config dict to a list
+        - Dump instance config
+        - Wait for SSH
```

Comment:
hm. Not sure if it's a good idea to convert these tasks into handlers. Can't it create bugs ?
Can others give their mood on this ? @ssbarnea  ?

Reply:
@wadhah101 

Is there any document about this enhancement that `handlers` is better than current implementation?

Reply:
The old config doesn't pass the default ansible-lint. I was following the ansible-lint and deepsource rules
https://ansible-lint.readthedocs.io/rules/no-handler/
https://app.deepsource.com/directory/analyzers/ansible/issues/ANS-E5003

I don't think it makes a big change in this template because all the handlers depend on a single task. I don't have experience encountering bugs with handlers but I am open to examples

Reply:
Thanks for your information. It's ok to use `handler`.

Besides, would you mind combining 4 handlers into one `block`? They are in one logical group.

Reply:
Handler currently do not support blocks https://github.com/ansible/ansible/issues/14270

The `listen` keyword seems to be the solution https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_handlers.html#naming-handlers

I have updated the code accordingly

Reply:
Could you change the format to the original one?

My original thought is to combine multiple tasks into one logic group via `block`. But unfortunately, `block` is not support in `handler`. Thanks for your information.

I am afraid the execution order of `nofity` runs unpredictably. 

There is an explicit statement in [official document about the following format](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_handlers.html#notifying-handlers) but no order explanation if we use `listen`.

```yaml
  notify:
    - Restart apache
    - Restart memcached
 ```

> Handlers are executed in the order they are defined in the handlers section, not in the order listed in the notify statement.

Besides, if we use `listen`, I verified, handlers are executed in the order listed in the notify statement.

Reply:
By original you mean without handlers or with notify ? @zhan9san

Reply:
With handler, and define multiple handlers in handler section